Fix core plugins and add new plugins and other changes

otiskujawa opened the pull request:
Make whois show newest history, not the oldest
Make help use footer instead of big image
Make alias for avatar

otiskujawa committed:
Merge pull request #2 from otiskujawa/patch-20

Patch 20

otiskujawa committed:
Merge pull request #3 from bluskript/work/blusk/christmas-image-processor

Work/blusk/christmas image processor

otiskujawa committed:
Fix help image

otiskujawa committed:
Make whois show newest history, not the oldest

otiskujawa committed:
add ytlink plugin and make it enabled only if token is present

otiskujawa committed:
add server banner plugin that somehow works

otiskujawa committed:
add user banner plugin that somehow doesn't work so disabled

otiskujawa commented:
whois before
![image](https://github.com/Geoxor/Naoko/assets/49094075/3d4501d4-0462-4a5d-9be4-862e4ac8ba9c)
after
![image](https://github.com/Geoxor/Naoko/assets/49094075/ff9a0d97-4122-4bce-ae77-19e43dafbaf2)

help before
![image](https://github.com/Geoxor/Naoko/assets/49094075/659ae488-4335-4d23-a976-04d6df750580)
after
![image](https://github.com/Geoxor/Naoko/assets/49094075/322adfef-7683-4209-8177-0df116a92f3d)

server banner command
![image](https://github.com/Geoxor/Naoko/assets/49094075/9ed2c823-b9b3-4841-bfe7-df671915cef3)

ytlink is disabled if no token is present
![image](https://github.com/Geoxor/Naoko/assets/49094075/873cfef8-81c0-4da0-a239-9438154b7531)

otiskujawa committed:
change how restricted channel works (used Its-treason code from comment on pull request 162)

otiskujawa commented:
changed https://github.com/Geoxor/Naoko/pull/162 using @Its-treason comment

otiskujawa commented:
also fixed #165

otiskujawa committed:
finnaly bump the version

otiskujawa commented:
ready to merge

Its-treason commented:
Hey @otiskujawa, pretty cool that you are still developing Naoko and I hope the rewrite I made earlier this year made it easier to change stuff around in Naoko.

I'm not sure if a review is really needed here, but found some smaller stuff after looking over the PR:

- Names of classes should be written in PascalCase (`svrbanner` to `SvrBanner`) and match the filename. Also, every time there is a name conflict, I suffixed the Command class with “Command” (`svrBanner` to `SvrBannerCommand`)
- There are some leftover debug `console.logs` and TODO's e.g., with the YT token

---

I'm also thinking about adding slash command support to Naoko. It should be possible to have slash commands and chat commands in parallel without much change. But I'm not sure if Geo wants to have slash commands. If Geo doesn't want slash commands, you could consider unregistering the current slash commands. Here's a small script made by ChatGTP:

```js
const { REST } = require('discord.js');
const { token } = require('./config.json');

const commands = [];

const rest = new REST({ version: '9' }).setToken(token);

(async () => {
  try {
    console.log('Started refreshing application (/) commands.');

    await rest.put(Routes.applicationCommands('<your_application_id>'), {
      body: commands,
    });

    console.log('Successfully removed all slash commands.');
  } catch (error) {
    console.error(error);
  }
})();
```

Its-treason committed:
Merge branch 'main' of github.com:otiskujawa/Shaii-otiskujawa into fix-various-bugs

Its-treason committed:
Switch from @triptyk/tsyringe to tsyringe and update readme

@triptyk/tsyringe was used before, because the main tsyringe package was not
  updated

Its-treason committed:
Allow returning a info in pipeline elements

Also update some texts and console.log's

Its-treason committed:
Fix MemorizeRoles trying give the everyone role

Its-treason committed:
Fix ~say command

Its-treason committed:
Fix ~unmute command

otiskujawa committed:
Merge pull request #4 from Its-treason/fix-various-bugs

Fix various bugs
